app/Dockerfile

```diff
@@ -1,12 +1,20 @@
 FROM node:22
 
+ARG PUBLIC_URL=
+ENV PUBLIC_URL=$PUBLIC_URL
+
+ARG REACT_APP_PACKIT_API_URL=
+
 COPY package.json /app/package.json
 COPY package-lock.json /app/package-lock.json
 WORKDIR /app
 RUN npm ci
 
 COPY . /app
 
+# override env var in production file if set
+RUN if [ -n "$REACT_APP_PACKIT_API_URL" ]; then  echo "REACT_APP_PACKIT_API_URL=$REACT_APP_PACKIT_API_URL" >> env.production.local;  fi
+
```

Comment:
I'm not sure why this only applies to production. It makes me uncomfortable to introduce differences between production and other environments, hence why I ask.

Reply:
It's "production" rather than "dev", so production npm build rather than say "production" vs "uat". This is forcing npm to override an environment variable used at build time (when it builds the docker image) - so it's the same issue with needing to build an image which will work for the Montagu deploy to get the right API url.